### PR TITLE
ICDS custom sms recipients and criteria

### PR DESCRIPTION
**Pull request by millerdev**
Review by commit.

@gcapalbo

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -1672,6 +1672,9 @@ def _determine_couch_databases(couch_databases):
     'ICDS_MOTHER_PERSON_CASE_FROM_CHILD_HEALTH_CASE':
         ['custom.icds.messaging.custom_recipients.recipient_mother_person_case_from_child_health_case',
          "ICDS: Mother person case from child_health case"],
+    'ICDS_MOTHER_PERSON_CASE_FROM_CHILD_PERSON_CASE':
+        ['custom.icds.messaging.custom_recipients.recipient_mother_person_case_from_child_person_case',
+         "ICDS: Mother person case from child case"],
```
Comment (gcapalbo) [inline settings.py]: nit: could we say "child **person** case"

Comment (gcapalbo) [reply settings.py]: nevermind, see you fixed this later

File: custom/icds/tests/test_custom_rules.py [hunk 2]
```diff
@@ -187,35 +188,46 @@ def _set_dob(self, case, dob):
...
+        self.assert_person_case_is_under_N_years_old(
+            19, 'ICDS_PERSON_CASE_IS_UNDER_19_YEARS_OLD')
+
+    def assert_person_case_is_under_N_years_old(self, n_years, criteria_name):
         rule = _create_empty_rule(self.domain, case_type='person')
-        rule.add_criteria(CustomMatchDefinition, name='ICDS_PERSON_CASE_IS_UNDER_6_YEARS_OLD')
+        rule.add_criteria(CustomMatchDefinition, name=criteria_name)
 
         with _with_case(self.domain, 'person', datetime.utcnow()) as case:
+            dob = datetime(2018, 2, 22, 12, 0)
```
Comment (gcapalbo) [inline custom/icds/tests/test_custom_rules.py]: this isn't the dob - this is the value for `now` that gets passed when the rule criteria evaluate

File: custom/icds/tests/test_custom_rules.py [hunk 3]
```diff
@@ -187,35 +188,46 @@ def _set_dob(self, case, dob):
...
             # No value for dob yet
-            self.assertFalse(rule.criteria_match(case, datetime(2018, 2, 22, 12, 0)))
+            self.assertFalse(rule.criteria_match(case, dob))
```
Comment (gcapalbo) [inline custom/icds/tests/test_custom_rules.py]: this isn't the dob - this is the value for `now` that gets passed when the rule criteria evaluate

Comment (gcapalbo) [reply custom/icds/tests/test_custom_rules.py]: nevermind, i see what you did below

File: custom/icds/tests/test_custom_rules.py [hunk 4]
```diff
@@ -187,35 +188,46 @@ def _set_dob(self, case, dob):
...
-            self.assertTrue(rule.criteria_match(case, datetime(2018, 5, 22, 12, 0)))
+            for assert_, age in [
+                # Test over N years old
+                (self.assertFalse, relativedelta(years=n_years)),
+                (self.assertFalse, relativedelta(years=n_years, days=90)),
+                # Test under N years old
+                (self.assertTrue, relativedelta(years=n_years, days=-1)),
+                (self.assertTrue, relativedelta(days=90)),
+            ]:
+                assert_(rule.criteria_match(case, dob + age), age)
```
Comment (gcapalbo) [inline custom/icds/tests/test_custom_rules.py]: there's an extra ", age)" at the end that shouldn't be there right?

Comment (millerdev) [reply custom/icds/tests/test_custom_rules.py]: that's so you can tell which one failed if the assertion fails

**Review by gcapalbo — APPROVED** [review]

**Pull request merged**